src/bson/bson.c

```diff
@@ -2728,9 +2728,16 @@ _bson_as_json_visit_regex (const bson_iter_t *iter,
 {
    bson_json_state_t *state = data;
    const char *c;
+   char *escaped_regex;
+
+   escaped_regex = bson_utf8_escape_for_json (v_regex, -1);
```

Comment:
As mentioned in my comment in [CDRIVER-2168](https://jira.mongodb.org/browse/CDRIVER-2168), this causes the [regex with slash](https://github.com/mongodb/specifications/blob/0fc09a2bc56de5fbee6f9c046083d39b5b62faf2/source/bson-corpus/tests/regex.json#L22) test from the BSON corpus suite to fail.

@xdg cited the [section 7 of RFC-7159](https://tools.ietf.org/html/rfc7159#section-7) in a related HELP ticket as evidence that `bson_utf8_escape_for_json()` need not do this. Additionally, [JSON.org](http://www.json.org/) suggests that escaping of forward slashes is optional.